File: app/src/main/java/com/aspire/dubsmash/siavash/ApplicationBase.java
package com.aspire.dubsmash.siavash;

import android.app.Application;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

import io.realm.Realm;

/**
 * Created by sia on 9/3/15.
 */
public class ApplicationBase extends Application {
    private static String sAppName;
    private static Realm sRealm;

    public static String getAppName() {
        return ApplicationBase.sAppName;
    }

    public static Realm getRealm() {
        return sRealm;
    }

    @Override public void onCreate() {
        super.onCreate();
        try {
            ApplicationInfo appInfo = getPackageManager().getApplicationInfo(getPackageName(), PackageManager.SIGNATURE_MATCH);
            sAppName = appInfo != null ? (String) getPackageManager().getApplicationLabel(appInfo) : "unknown";
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

        sRealm = Realm.getInstance(this);
    }
}

File: app/src/main/kotlin/com.aspire.dubsmash/ApplicationBase.kt
package com.aspire.dubsmash

import android.app.Application
import android.content.pm.PackageManager
import io.realm.Realm

/**
 * Created by sia on 11/19/15.
 */
class ApplicationBase : Application() {
    companion object ApplicationName {
        lateinit var appName: String
        lateinit var realm : Realm
    }

    override fun onCreate() {
        super.onCreate()

        val appInfo = packageManager.getApplicationInfo(packageName, PackageManager.SIGNATURE_MATCH)
        appName = packageManager.getApplicationLabel(appInfo) as String

        realm = Realm.getInstance(this)

//        Fabric.with(this, Crashlytics());
    }
}
